src/libs/filters/aperture.c

```diff
@@ -67,7 +67,19 @@ static gboolean _aperture_update(dt_lib_filtering_rule_t *rule)
 
 static gchar *_aperture_print_func(const double value, const gboolean detailled)
 {
-  return g_strdup_printf("%s%.1lf", detailled ? "f/" : "", value);
+  if(detailled)
+  {
+    return g_strdup_printf("%s%.1lf", detailled ? "f/" : "", value);
+  }
+  else
+  {
+    gchar *locale = g_strdup(setlocale(LC_ALL, NULL));
+    setlocale(LC_NUMERIC, "C");
```

Comment:
setlocale is not thread-safe, this is not a correct way to solve this.

Reply:
All this is happening from the main Gtk GUI thread as far as I understand, we should be safe, no?

Reply:
In C++ one can use stringstream and imbue() to set locale just for that stream. I’m not sure if anything simular exists in C. My suggestion is to create a helper function (to be used from C code) to format floating point number with dot separator and implement it in C++.

Reply:
> All this is happening from the main Gtk GUI thread as far as I understand, we should be safe, no?

Are you sure no other threads use functions like printf? Even threads created by libraries?

No, it’s not safe.

Reply:
Right, just after my message it occurs to me that some export could be running in background and so we will change the local also for them.

Reply:
Ok, then we should use `duplocale` and `newlocale`.
This is needed for the exposure filter as well.

Reply:
Right, uselocale is a C-way to solve this.
And FYI this is a kind of issues created by setlocale: https://github.com/darktable-org/darktable/issues/13637#issuecomment-1431006172

Reply:
Also be advised that on Windows it may require different code, see linked lensfun issue.

Reply:
https://github.com/lensfun/lensfun/pull/1987/files

Reply:
I have now created a function in `common/utility.c`.
Tested on Windows as well.

BTW: There are several other calls of `setlocale` in the code base which should be adapted.

Reply:
Fixed the other filters as well: exposure, iso, ratio